### PR TITLE
chore(DEV-1113): Add GitHub action to validate PR title to meet semantic versioning patterns

### PR DESCRIPTION
**Pull request by jazeee**
Summary
===
The semantic versioning relies on the title of the commit to determine
whether the next semantic version release.

This change is intended to solve the problem that `commitlint` attempts
to solve.
Unfortunately, `commitlint` attempts to sanitize commit messages, before
the PR is created or updated.
It doesn't work, and encourages bad patterns and workarounds like
using a nonsensical commit message like:
`fix: Ab cd` or `docs: He ll`

Worse, it doesn't allow us to add useful context like long URLs.
Reference:

https://www.notion.so/0019-Commit-Messages-b7cc6f56354f4f28b902df7cd13b2450

In addition, people don't typically review the commit message, nor
does anyone fix the commit message content. The commit messages remain
uncurated.

In contrast, the PR title and description are read by reviewers, and are
more
likely to be curated.
In addition, since the PR title and description is better curated,
the PR makes sense as the source of truth for the bolus of change.
In other words, it is the best piece of information to drive the commit
log.

Reference:

https://linear.app/clipboardhealth/issue/DEV-1113/add-semantic-version-github-action-to-cbh-core

Changes
===
- Add a github action to check the PR title

Video
---
Tested with invalid prefixes like `shore`, which was caught as invalid.
https://www.loom.com/share/d03f8499171c42c3ad77026637d7ebe0

### CODE REVIEW

File: .github/workflows/pull-request-title.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: PR Title
```
Comment (jazeee) [inline .github/workflows/pull-request-title.yml:1]: It's unclear if this should be in `/.github/` directory or the parent.
How does this special repo `.github` apply its contents?
I didn't find any docs from a scan online. (Of course, its not an easy search)

This file does work, for this repo:
https://github.com/ClipboardHealth/.github/actions/runs/7509833906/job/20447464365?pr=3

I suspect though, that we want to move this file from `/.github/workflows/` to `/workflows/`

**Review by therockstorm — APPROVED** [review]

File: .github/workflows/pull-request-title.yml (hunk 1 above)
Comment (pgbezerra) [reply .github/workflows/pull-request-title.yml:1]: You can find your answer by reading this doc: https://docs.github.com/en/actions/using-workflows/creating-starter-workflows-for-your-organization

It must be in the `workflow-templates` folder, and have a properties file.

Comment (jazeee) [reply .github/workflows/pull-request-title.yml:1]: Ok, I'll move it. I'll also add a readme.
